pkg/downwardmetrics/virtio-serial/server.go

```diff
@@ -54,8 +55,8 @@ const (
 // (will also fail for `maxRequestsBurst` > 256)
 const _ = uint8(maxRequestsBurst - 1)
 
-func RunDownwardMetricsVirtioServer(ctx context.Context, nodeName, channelSocketPath, launcherSocketPath string) error {
-	report, err := newMetricsReporter(nodeName, launcherSocketPath)
+func RunDownwardMetricsVirtioServer(ctx context.Context, signalStopChan chan struct{}, nodeName, channelSocketPath string) error {
```

Comment:
If you pass signalStopChan I don't think there's a point in passing the context anymore. Both are used to stop the server in a controlled manner.

Reply:
Context is still required to ratelimit the server.

Reply:
Done.